ingenialink/servo.py

```diff
@@ -1284,8 +1286,45 @@ def read(
         if _reg.dtype == REG_DTYPE.BYTE_ARRAY_512:
             return raw_read
         value = convert_bytes_to_dtype(raw_read, _reg.dtype)
+        self._notify_register_update(_reg, value)
         return value
 
+    def register_update_subscribe(self, callback: Callable[["Servo", Register], None]) -> None:
+        """Subscribe to register updates.
+        The callback will be called when a read/write operation occurs.
+
+        Args:
+            callback: Callable that takes a Servo and a Register instance as arguments.
+
+        """
+        self.__register_update_observers.append(callback)
+
+    def register_update_unsubscribe(self, callback: Callable[["Servo", Register], None]) -> None:
+        """Unsubscribe to register updates.
+
+        Args:
+            callback: Subscribed callback.
+
+        """
+        self.__register_update_observers.remove(callback)
+
+    def _notify_register_update(self, reg: Register, data: Union[int, float, str, bytes]) -> None:
+        """Notify a register update to the observers.
+        The updated value is stored in the register's storage attribute.
+
+        Args:
+            reg: Updated register.
+            data: Updated value.
+
+        """
+        reg.storage_valid = True
+        reg.storage = data
```

Comment:
Talking with @polfeliu , we thought that the storage attribute in the register should not be used. Dictionary and Register classes should be static. For the moment we don't need a cache for the registers values, and if we need it in the future maybe the registers are not the best class to implement it.

Reply:
Done